src/test/groovy/ratpack/zipkin/internal/ZipkinHttpClientImplSpec.groovy

```diff
@@ -87,8 +88,30 @@ class ZipkinHttpClientImplSpec extends Specification {
 			HttpMethod.DELETE | _
 			HttpMethod.HEAD | _
 			HttpMethod.OPTIONS | _
+	}
+
+	def "Request should not duplicate headers sent"() {
+		given:
+			webServer.enqueue(new MockResponse().setResponseCode(200))
 
+		when:
+			harness.yield { e ->
+				harnessSetup(e)
+				zipkinHttpClient.request(uri, {spec ->
+					spec.method(HttpMethod.GET).headers.add("X-TEST", "test")
+				})
+			}
+
+		then:
+			reporter.spans.size() == 1
+			Span span = reporter.spans.get(0)
+			span.name == "get"
+
+		then:
+			RecordedRequest request = webServer.takeRequest()
+			request.headers.values("X-TEST").size() == 1
```

Comment:
Before the change in `ZipkinHttpClientImpl` in this PR it would return 2 headers for `X-TEST`.

Reply:
Like the test